spark/src/main/scala/geotrellis/spark/op/local/spatial/LocalSpatialRasterRDDMethods.scala

```diff
@@ -0,0 +1,24 @@
+package geotrellis.spark.op.local.spatial
+
+import geotrellis.raster.op.local._
+import geotrellis.spark._
+import geotrellis.vector.Geometry
+
+abstract class LocalSpatialRasterRDDMethods[K: SpatialComponent] extends RasterRDDMethods[K] with Serializable {
+
+  /** Masks this raster by the given Geometry. */
+  def mask(geom: Geometry): RasterRDD[K] = {
+    mask(Seq(geom))
+  }
+
+  /** Masks this raster by the given Geometry. */
+  def mask(geoms: Iterable[Geometry]): RasterRDD[K] = {
+    val mapTransform = rasterRDD.metaData.mapTransform
+    rasterRDD.mapPairs { case (k, tile) =>
+      val key = k.spatialComponent
+      val result  = tile.mask(mapTransform(key), geoms)
```

Comment:
Let's go a different route....

Let's make mask work with Polygon and MultiPolygon. That's really the only types we'll be masking with, so it's better to be explicit about that. Extent can be transformed into a Polygon so that takes care of that case.

If we wrote a mask for Polygon and MultiPolygon, we could do one of two things: always wrap the polygon to a MultiPolygon, or write two identical looking versions. Depends on how much code duplication that would accumulate, but I'd almost prefer the explicit duplication (because wrapping always leads to worse performance).

I'm wondering if we want to write a mask also for Extent that doesn't even use the Rasterizer, but uses a RasterExtent to determine what cells to include (which would be waaay faster). I think that might be the way to go.